test/set.js

```diff
@@ -22,6 +22,22 @@ describe('set', function() {
     assert.deepEqual(b, {w: 1, x: {a: 5, b: 2, c: 3}, z: 4});
   });
 
+  it('handles empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({}, a);
+    assert.deepEqual(b, a);
+  });
+
+  it('handles nested empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({w: 0, x: {}, z: 3}, a);
+    assert.deepEqual(b, {w: 0, x: {a: 1, b: 2, c: 3}, z: 3});
+  });
+
```

Comment:
Well, I wasn't so sure I bought the previous test either.  But this one was easier to discuss.

This change will mean that you can't even do the following:

```js
    set({x: 42, z: 100}, {w: 1, x: {a: 2, b: 3}, y: 4}); 
    //=> want {w: 1, x: 42, y: 4, z: 100},
    //   get  {w: 1, x: {a: 2, b: 3}, y, 4, z: 100}
```

Before this change, that worked as expected.  You could replace whole objects with a new value.

But even if we fix that, we're left with

```js
    var a =  {w: 1, x: {a: 2, b: 3}, y: 4};
    var b = set({x: 42}, a)
    var c = set({x: {}}, b); //=> {w: 1, x: {}, y: 4}
    
    var d = set({x: {}}, a); //=> {w: 1, x: {a: 2, b: 3}, y: 4};
```

Which seems rather bizarre.

I did warn you that I thought there were prickers in deep merging!  :smiling_imp: